src/server/vimeo/client.ts

```diff
@@ -0,0 +1,54 @@
+import { OAuth2Client } from 'google-auth-library';
+import { google, youtube_v3 } from 'googleapis';
+import { Readable } from 'node:stream';
+import sanitize from 'sanitize-html';
+import { getEdgeUrl } from '~/client-utils/cf-images-utils';
+import { fetchBlob } from '~/utils/file-utils';
+
+type S3ToVimeoUpload = {
+  url: string;
+  title: string;
+  description: string;
+  mimeType?: string;
+  accessToken: string;
+  size: number;
+};
+
+export const uploadVimeoVideo = async ({
+  url,
+  accessToken,
+  title,
+  description,
+  size,
+}: S3ToVimeoUpload) => {
+  const res = await fetch(`https://api.vimeo.com/me/videos`, {
```

Comment:
Should the fetch endpoint come from the env vars?
```suggestion
  const res = await fetch(env.VIMEO_VIDEO_UPLOAD_URL, {
```

Reply:
No, the upload URL is in the case we had a worker in place, which as it turns out we don't really need apparently 🗡️